From 469b43ee1f1836c319c62c79136462cf0ab04f44 Mon Sep 17 00:00:00 2001
From: Liang Chen <cl@rock-chips.com>
Date: Tue, 24 Sep 2019 11:33:15 +0800
Subject: [PATCH] fiq_debugger: fix bug when define CONFIG_THREAD_INFO_IN_TASK

when define CONFIG_THREAD_INFO_IN_TASK, thread_info is in the task_struct,
and current_tread_info() is always point to the right current thread info
in irq/fiq mode, so do not change *current_tread_info().

Change-Id: Ic6476efc5f13cedb26405fd56646ee7c939dff45
Signed-off-by: Liang Chen <cl@rock-chips.com>
---
 drivers/staging/android/fiq_debugger/fiq_debugger_arm64.c | 2 --
 drivers/staging/android/fiq_debugger/fiq_debugger_priv.h  | 2 ++
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/staging/android/fiq_debugger/fiq_debugger_arm64.c b/drivers/staging/android/fiq_debugger/fiq_debugger_arm64.c
index c53f4980bab9..190aa0f1ac28 100644
--- a/drivers/staging/android/fiq_debugger/fiq_debugger_arm64.c
+++ b/drivers/staging/android/fiq_debugger/fiq_debugger_arm64.c
@@ -177,12 +177,10 @@ static int report_trace(struct stackframe *frame, void *d)
 void fiq_debugger_dump_stacktrace(struct fiq_debugger_output *output,
 		const struct pt_regs *regs, unsigned int depth, void *ssp)
 {
-	struct thread_info *real_thread_info = THREAD_INFO(ssp);
 	struct stacktrace_state sts;
 
 	sts.depth = depth;
 	sts.output = output;
-	*current_thread_info() = *real_thread_info;
 
 	if (!current)
 		output->printf(output, "current NULL\n");
diff --git a/drivers/staging/android/fiq_debugger/fiq_debugger_priv.h b/drivers/staging/android/fiq_debugger/fiq_debugger_priv.h
index d5d051f727a8..915860a6b38a 100644
--- a/drivers/staging/android/fiq_debugger/fiq_debugger_priv.h
+++ b/drivers/staging/android/fiq_debugger/fiq_debugger_priv.h
@@ -16,8 +16,10 @@
 #ifndef _FIQ_DEBUGGER_PRIV_H_
 #define _FIQ_DEBUGGER_PRIV_H_
 
+#ifndef CONFIG_THREAD_INFO_IN_TASK
 #define THREAD_INFO(sp) ((struct thread_info *) \
 		((unsigned long)(sp) & ~(THREAD_SIZE - 1)))
+#endif
 
 struct fiq_debugger_output {
 	void (*printf)(struct fiq_debugger_output *output, const char *fmt, ...);
-- 
2.35.3

